docker/build/devdeps.manylinux.Dockerfile

```diff
@@ -63,21 +63,27 @@ RUN if [ "$toolchain" == 'gcc11' ]; then \
 ENV CC="$LLVM_INSTALL_PREFIX/bootstrap/cc"
 ENV CXX="$LLVM_INSTALL_PREFIX/bootstrap/cxx"
 
-# Build the the LLVM libraries and compiler toolchain needed to build CUDA Quantum
-ADD ./scripts/build_llvm.sh /scripts/build_llvm.sh
-ENV LLVM_BUILD_LINKER_FLAGS="-static-libgcc -static-libstdc++"
+# Build pybind11 - 
+# we should be able to use the same pybind version independent on what Python version we generate bindings for.
 RUN dnf install -y --nobest --setopt=install_weak_deps=False \
-        ninja-build cmake \
+        ninja-build cmake python3-devel \
     && mkdir /pybind11-project && cd /pybind11-project && git init \
     && git remote add origin https://github.com/pybind/pybind11 \
     && git fetch origin --depth=1 $pybind11_commit && git reset --hard FETCH_HEAD \
     && mkdir -p /pybind11-project/build && cd /pybind11-project/build \
-    && cmake -G Ninja ../ -DCMAKE_INSTALL_PREFIX=/usr/local/pybind11 \
+    && python3 -m ensurepip --upgrade && python3 -m pip install pytest \
+    && cmake -G Ninja ../ -DCMAKE_INSTALL_PREFIX=/usr/local/pybind11 -DPYTHON_EXECUTABLE="$(which python3)" \
     && cmake --build . --target install --config Release \
-    && cd .. && rm -rf /pybind11-project \
-    && export CMAKE_EXE_LINKER_FLAGS="$LLVM_BUILD_LINKER_FLAGS" CMAKE_SHARED_LINKER_FLAGS="$LLVM_BUILD_LINKER_FLAGS" \
-    && bash /scripts/build_llvm.sh -s /llvm-project -c Release -v 
-    # No clean up, since we need to re-build llvm for each python version to get the bindings.
+    && python3 -m pip uninstall -y pytest \
```

Comment:
Why do we want to uninstall `pytest` here?

Reply:
Why not? It is no longer needed, and this image is not one that people should use for development. I uninstalled it because the image container 5 or 6 python versions; subsequent steps need to be done for each version and that mean also all necessary pip packages will need to be installed for that version.

Reply:
> It is no longer needed, and this image is not one that people should use for development.

OK, if we don't want people to use this image for development, then that sounds reasonable. It just seems like a tiny package that is useful for testing, so I figured I'd ask.